mlir/test/Dialect/Linalg/transform-op-decompose.mlir

```diff
@@ -250,6 +282,8 @@ module attributes {transform.with_named_sequence} {
 
     %2 = transform.structured.match ops{["linalg.softmax"]} in %arg1 : (!transform.any_op) -> !transform.any_op
     %3 = transform.structured.decompose_interface %2 : (!transform.any_op) -> !transform.any_op
+    %4 = transform.structured.generalize %3: (!transform.any_op) -> !transform.any_op
```

Comment:
Why add this to the existing test?
I don't think generalization is needed for anything and it makes it harder to see what, if anything, has changed after the logic rework.

Reply:
This is to show that the generic lowering is still valid. The lowering to named ops is in another test.

Reply:
> generic lowering is still valid

It used to be decomposed to generic but now the decomposition is always to named ops and you have to rely on separate pass to have that representation. So, I still think it's unrelated to this test.

Reply:
I think we need both. It doesn't matter to me if we create a new one for generic or rely on the new one Petr added.

Reply:
To me it's a separate concern. AFAIK, decompose has never promised specific lowering format.
If we worry about generalization pass then that belongs to its own unit tests.

Reply:
I think it's a nice demonstration of the resulting code which is useful for this review. I agree it's not related to the test. We can remove generalization from both decomposition and transform tests before the merge.

Reply:
I agree with @rengolin that we need both. MLIR could definitely use more of "integration" tests, but without them replacing unit tests.

Reply:
I'm confused. Does "both" mean we want to leave all the existing tests in the PR and add one more test for the transform op without generalization? Like in:
```mlir
module attributes {transform.with_named_sequence} {
  transform.named_sequence @__transform_main(%arg1: !transform.any_op {transform.readonly}) {
    %2 = transform.structured.match ops{["linalg.softmax"]} in %arg1 : (!transform.any_op) -> !transform.any_op
    %3 = transform.structured.decompose_interface %2 : (!transform.any_op) -> (!transform.any_op)
    transform.yield
  }
}
```

Reply:
To me you already have both. The old test, which you generalize and compare the delta and the new one you added, which is just the named ops. I'm happy with the tests the way they are.

Reply:
I agree with what's being proposed here (in terms of test coverage), but please make sure that we don't duplicate tests and that test files reflect what's being tested (I consider test file names to be part of documentation). 

ATM:
1. This file is called "transform-op-decompose.mlir", but it's being updated to test both decomposition _and_ generalisation. 
2. Another file added in this PR, "decompose-named-ops.mlir", also tests both decomposition _and_ generalisation.

IMO, there's should be a separate test file for:
* decomposition
* generalisation
* e2e/integration (decomposition -> generalisation

This way it will be clear _what_ is being tested and _where_.